sources/packages/web/src/views/student/StudentProfile.vue

```diff
@@ -56,15 +99,53 @@ export default {
     const store = useStore();
     const router = useRouter();
     const toast = useToast();
+    const showApplyPDButton = ref();
+    const disableBtn = ref(false);
     const initialData = ref({} as StudentFormData);
+    const studentAllInfo = ref({} as StudentFormInfo);
+    const getStudentInfo = async () => {
+      studentAllInfo.value = await StudentService.shared.getStudentInfo();
+    };
+    const appliedPDButton = async () => {
+      showApplyPDButton.value = false;
+      if (
+        studentAllInfo.value?.validSin &&
+        studentAllInfo.value?.pdSentDate === null &&
+        studentAllInfo.value?.pdVerified === null
+      ) {
+        showApplyPDButton.value = true;
+      }
+    };
+    const applyPDStatus = async () => {
+      disableBtn.value = true;
+      try {
+        await StudentService.shared.applyForPDStatus();
+        toast.add({
+          severity: "success",
+          summary: `Applied for PD Status!`,
+          detail: " Successfully!",
+          life: 5000,
+        });
+      } catch (error) {
+        toast.add({
+          severity: "error",
+          summary: "Unexpected error",
+          detail:
+            "An error happened during the apply PD process. Please try after sometime.",
+          life: 5000,
+        });
+      }
+      await getStudentInfo();
```

Comment:
The API calls could happen in parallel here. Please consider changing it.

Reply:
actually, `getStudentInfo` function update `studentAllInfo` variable  and this variable is used in `appliedPDButton` function, that why, I added await,

Reply:
await must be here but, since there is no dependency between getStudentInfo and appliedPDButton you can do something like:
```
const getStudentInfoResult = getStudentInfo(); 
const appliedPDButtonResult = appliedPDButton();
await Promise.All([getStudentInfoResult , appliedPDButtonResult ]);
```